plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseTaskStore.ts

```diff
@@ -85,6 +85,35 @@ export class DatabaseTaskStore implements TaskStore {
     this.db = options.database;
   }
 
+  async list(options: Partial<SerializedTask>): Promise<SerializedTask[]> {
+    const queryBuilder = this.db<RawDbTaskRow>('tasks');
+
+    if (options.createdBy) {
+      queryBuilder.where({
+        created_by: options.createdBy,
+      });
+    }
+
+    const results = await queryBuilder.orderBy('created_at', 'desc').select();
+
+    return results.map(result => ({
+      id: result.id,
+      spec: JSON.parse(result.spec),
+      status: result.status,
+      createdBy: result.created_by ?? undefined,
+      lastHeartbeatAt:
```

Comment:
Perhaps a small private function that does the date check/convertion as it is repeated four times?